python/graphstorm/graphpartitioning/metadata_schema.py

```diff
@@ -0,0 +1,799 @@
+                Each node type in this dictionary is expected to non-zero
+                features,
+        * ``edge_data``, if present, follows the same rules as described for
+                ``node_data`` in the earlier bullet point.
+
+    Examples:
+    ---------
+    Creating and initalizaing the singleton instance during the pipelines
+    initialization process.
+    >>> obj = Metadata()
+    >>> base_dir = "/home/ubuntu/graph-dataset"
+    >>> file_path = "metadata.json"
+    >>> obj.init(file_path, base_dir)
+
+    After initialization is completed, general usage of any method or property
+    in this class is
+    >>> ntypes = Metadata().ntypes # To retrieve node types from the graph
+    >>> # To get the id <-> ntype map
+    >>> # id_ntype_map will be {0: ``ntype0``, 1: ``ntype1``, ...}
+    >>> id_ntype_map = Metadata().id_ntype_map
+    """
+
+    _instance = None
+
+    def __new__(cls):
+        if cls._instance is None:
+            cls._instance = super(MetadataSchema, cls).__new__(cls)
+        return cls._instance
+
+    def __init__(self):
+        if self._instance is not None:
+            return
+
+        self._basedir = None
+        self._metadata_path = None
+        self._data = None
+        self._global_nid_offsets = None
+        self._global_eid_offsets = None
+        self._ntypes = None
+        self._ntype_id_map = None
+        self._id_ntype_map = None
+        self._etype_id_map = None
+        self._id_etype_map = None
+        self._etypes = None
+        self._ntype_features = None
+        self._etype_features = None
+        self._ntype_feature_files = None
+        self._etype_feature_files = None
+        self._etype_files = None
+
+    def _load_json(self, file_name, base_dir):
+        """Helpper function to read the metadata schema file of the input
+        graph dataset. ``file_name`` input argument can be either in absolute
+        or relative path formats. ``base_dir`` is used as the root directory
+        for the input graph dataset where all the files (of the dataset) are
+        expected to be located on the disk.
+
+        Argument
+        --------
+        file_name: string
+            Filename of the input metadata schema file. This can be either
+            in absolute or relative path formats. If in relative format,
+            ``base_dir`` is used as the root directory.
+        base_dir: string
+            Base directory, if provided, is used as the root directory for
+            the input graph dataset
+        """
+        if file_name is None or not isinstance(file_name, str):
+            raise ValueError(f"File path is not valid. = {file_name}")
+
+        self._basedir = None
+        if base_dir is not None and os.path.isdir(base_dir):
+            if not os.path.isfile(os.path.join(base_dir, file_name)):
+                raise ValueError(
+                    f"Metadata file does not exist. {base_dir}, {file_name}"
+                )
```

Comment:
Why raise a `ValueError` here instead of letting Python raise a more appropriate `FileNotFoundError`?

Reply:
The line of thought was that the function arguments are not valid in this case and hence the code.

Modified the code to let the system handle it appropriately.